packages/amplify-cli-core/src/help/help-helpers.ts

```diff
@@ -0,0 +1,236 @@
+  subCommands: Array<SubCommandInfo>
+};
+
+const SPACE = ' ';
+const DEFAULT_COLUMN_SEP_SPACING = 2;
+const DEFAULT_INDENT_SPACING = 2;
+const DEFAULT_COLUMN_SEP = SPACE.repeat(DEFAULT_COLUMN_SEP_SPACING);
+const DEFAULT_INDENT = SPACE.repeat(DEFAULT_INDENT_SPACING);
+
+const USAGE = 'USAGE';
+const DESCRIPTION = 'DESCRIPTION';
+const COMMANDS = 'COMMANDS';
+const FLAGS = 'FLAGS';
+const LEARN_MORE = 'LEARN MORE';
+const SUBCOMMANDS = 'SUBCOMMANDS';
+const TAG_LINE = 'The Amplify Command Line Interface (CLI) is a unified toolchain to create, integrate, and manage the AWS cloud services for your app.';
+const DEFAULT_LINK = 'https://docs.amplify.aws/cli/';
+
+const FLAG_DELIMITER = ' | ';
+const FLAG_PREFIX_SHORT = '-';
+const FLAG_PREFIX_LONG = '--';
+
+const printHeaderText = (text: string) => printer.info(chalk.blue.bold(text));
+const printBodyText = (text: string) => printer.info(text);
+const printCategoryMessage = () => {
+  printBodyText(DEFAULT_INDENT + 'Where <category> is one of: notifications, api, auth, custom, storage,');
+  printBodyText(DEFAULT_INDENT + 'analytics, function, geo, hosting, interactions, predictions, xr');
+  printer.blankLine();
+};
+
+export function lookUpCommand(commandsInfo: Array<CommandInfo>, commandName: string): CommandInfo | null {
+  let retCommand: CommandInfo | null = null;
+  for (const command of commandsInfo) {
+    if (command.command == commandName) {
+      retCommand = command;
+      break;
+    }
+  }
+  return retCommand;
+}
+
+export function lookUpSubcommand(commandsInfo: Array<CommandInfo>, commandName: string, subcommandName: string): SubCommandInfo | null {
+  let command = lookUpCommand(commandsInfo, commandName);
+  if (command === null) {
+    return null;
+  }
+  let retSubcommand: SubCommandInfo | null = null;
+  for (const subCommand of command.subCommands) {
+    if (subCommand.subCommand === subcommandName) {
+      retSubcommand = subCommand;
+      break;
+    }
+  }
+  return retSubcommand; 
+}
+
+export function parseHelpCommands(input: $TSAny, commandsInfo: Array<CommandInfo>) {
+  // depending on the commands and the order of commands, information is stored in different fields of Input
+  let specifiedCommands = {command: '', subCommand: ''};
+    // get all allowed commands/subcommands from json object
+    let acceptableCommands: Array<string> = [];
+    commandsInfo.forEach(command => acceptableCommands.push(command.command));
+    commandsInfo.forEach(command => command.subCommands.forEach(subCommand => acceptableCommands.push(subCommand.subCommand)));
+    let hasSubcommands = input.subCommands! && Array.isArray(input.subCommands) && input.subCommands.length >= 1; // check if subcommands exist
+    if (hasSubcommands) {
+      specifiedCommands = {command: input.subCommands[0], subCommand: ''}; // if just 1 subcommand, set that as command
+      if (input.subCommands.length == 1) {
+        if (input.options!) { // check if subcommands are in options field
```

Comment:
This non-null assertion does not have any effect, should be removed.

Reply:
Thanks